### ISSUE TITLE
GeneralizedTime is not C-OER compliant

### ISSUE THREAD

**Issue by sublimator** [interledgerjs/ilp-packet#37]
Relevant to:
https://github.com/interledger/rfcs/issues/440

See impl: 
https://github.com/interledgerjs/ilp-packet/blob/master/src/utils/date.ts#L43

Considering:
```
> new Date().toISOString()
'2018-07-16T08:37:52.250Z'
```

Non significant 0s and decimal points should be stripped from the microseconds at the least ? ( `at the least` meaning I'm not entirely clear about all the rules, but have noticed some by playing briefly with the encoding features of ASN.1 studio from OSS.com )

**Comment by sublimator** [comment]
Some relevant docs:
[X.690-201508](https://www.itu.int/rec/dologin_pub.asp?lang=e&id=T-REC-X.690-201508-I!!PDF-E&type=items) (DER/BER/CER)
[X.696-201508](https://www.itu.int/rec/dologin_pub.asp?lang=e&id=T-REC-X.696-201508-I!!PDF-E&type=items) (OER/C-OER)

**Comment by adrianhopebailie** [comment]
I don't think `GeneralizedTime` is used anywhere anymore. If it is then that is likely a bug.

That function should probably be marked as deprecated and removed soon to avoid confusion.

**Comment by sublimator** [comment]
Yeah, ILP is a bit hard to follow with all the churn. I understand that it's mostly quietened down but it seems the waters have yet to clear. 

Functions using GeneralizedTime:
```
deserializeIlqpLiquidityResponse
serializeIlqpLiquidityResponse
serializeIlpError
deserializeIlpError
```
I understand IlpError is from 'legacy' ILP and ILQP is from the quoting protocol. Whatever that means ... 

How long will they be deprecated for before being dropped ? Does being deprecated preclude bug fixes ?

**Comment by sublimator** [comment]
@adrianhopebailie 

I see a commit with with the message [add asn for ilp4](https://github.com/interledger/rfcs/commit/38fe2974584face253d3d6004bf32dc0342ae845)

The InterLedgerPacket definition still contains [InterledgerProtocolErrorData](https://github.com/interledger/rfcs/blob/master/asn1/InterledgerPacket.asn#L45)  which [is using GeneralizedTime](https://github.com/interledger/rfcs/blob/master/asn1/LegacyInterledgerProtocol.asn#L49).

What's a good resource for the short of time for determining the status of these protocols? You mentioned some are deprecated.

**Comment by emschwartz** [comment]
ILPv4 only includes `IlpPrepare`, `IlpFulfill`, and `IlpReject` so all of the others should be considered deprecated.

**Comment by sublimator** [comment]
Thanks Evan.

Is anyone still using ILP != 4?

**Comment by emschwartz** [comment]
Not on the live network at least (cause connectors won't forward those packets)

**Comment by sublimator** [comment]
I'll say from a noob perspective the 'legacy' naming seems to indicate that it's still supported. ILP might be more approachable if the past was swept away somewhere

`rfcs/deprecated` or something :)

Although, I'm coming at this perhaps from an unusual direction, picking a hyperledger/quilt issue as entry point: https://github.com/hyperledger/quilt/issues/142

**Comment by adrianhopebailie** [comment]
I'm busy reviewing a lot of the code and docs in the interledger/rfcs and interledgejs/ilp-core etc repos to weed out old stuff. It's very confusing for anyone new.

Actually discussed this with @justmoon yesterday and planning to do a good clean out. Your point is well taken 👍

**Comment by sublimator** [comment]
Nice :)
